### PR TITLE
Activities rework

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/Activities-rework-e727f61ebffb4667aa5b000142b86ef2

### CODE REVIEW

File: src/store/wallet/wallet-epics.ts [hunk 1]
```diff
@@ -133,25 +124,25 @@ const waitForOperationCompletionEpic = (action$: Observable<Action>, state$: Obs
...
-      ]).pipe(
-        map(([operations, fa12Operations, fa2Operations, transfers]) =>
-          groupActivitiesByHash(operations, fa12Operations, fa2Operations, transfers)
-        ),
-        map(activityGroups => loadActivityGroupsActions.success(activityGroups)),
-        catchError(err => of(loadActivityGroupsActions.fail(err.message)))
-      )
-    )
-  );
+// const loadActivityGroupsEpic = (action$: Observable<Action>, state$: Observable<RootState>) =>
```
Comment (LendiHop) [inline src/store/wallet/wallet-epics.ts:127]: ?

File: src/components/activity-groups-list/activity-groups-list.tsx [hunk 2]
```diff
@@ -39,18 +41,22 @@ export const ActivityGroupsList: FC<Props> = ({ activityGroups }) => {
...
-      renderItem={({ item }) => <ActivityGroupItem group={item} />}
-      renderSectionHeader={({ section: { title } }) => <Text style={styles.sectionHeaderText}>{title}</Text>}
-    />
+    <>
+      <SectionList
+        sections={sections}
+        stickySectionHeadersEnabled={true}
+        contentContainerStyle={styles.sectionListContentContainer}
+        onEndReachedThreshold={0.01}
+        onEndReached={loadMore}
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-groups-list.tsx:55]: you have got 3 different names `onEndReached`, `loadMore`, `handleUpdate`.  Rename `loadMore` to reduce different naming

File: src/hooks/use-general-activity.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      if (lastActivityGroup.length > 0) {
+        setLastId(lastActivityGroup[0].id ?? null);
+      }
+    }
+  };
+
+  return {
+    handleUpdate,
+    activities,
+    isAllLoaded
```
Comment (unixvb) [inline src/hooks/use-general-activity.ts:79]: `isAllLoaded` is never used outside the hook, but the change of `isAllLoaded` => will trigger the change of hook returned object => will trigger the change of `activities` => will trigger rerender.  It's better to remove it

Comment (unixvb) [reply src/hooks/use-general-activity.ts:79]: the same in other hooks

File: src/utils/token-operations.util.ts [hunk 4]
```diff
@@ -1,57 +1,46 @@
-import { from } from 'rxjs';
-import { map } from 'rxjs/operators';
-
 import { tzktApi } from '../api.service';
+import { OPERATION_LIMIT } from '../config/general';
```
Comment (unixvb) [inline src/utils/token-operations.util.ts:2]: You are using only `data` property of the returned value of functions below.   
So, you could map the result in this file and return prepared `data`

File: src/hooks/use-general-activity.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    if (outcomingOperations.data.length < 0) {
+      return;
+    }
+    const localLastItem = outcomingOperations.data[outcomingOperations.data.length - 1];
+    if (!isDefined(localLastItem)) {
+      setIsAllLoaded(true);
+
+      return;
+    }
+    const localLastId = localLastItem.id;
```
Comment (unixvb) [inline src/hooks/use-general-activity.ts:38]: rename to upperId & lowerId

**Review by unixvb — APPROVED** [review]

**Pull request merged**